code/datums/components/creamed.dm

```diff
@@ -10,7 +10,7 @@ GLOBAL_LIST_INIT(creamable, typecacheof(list(
   * For when you have pie on your face
   */
 /datum/component/creamed
-	dupe_mode = COMPONENT_DUPE_UNIQUE
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS

```

Comment:
There aren't any init args to pass along, what are you trying to do here? You also haven't defined the inherit proc which is what's needed to make use of this dupe_mode

Reply:
Alright, problem with COMPONENT_DUPE_UNIQUE is that when the new compoment is deleted, it also cleans parent face, which caused the issue above. By using COMPONENT_DUPE_UNIQUE_PASSARGS, even without InheritComponent or init args, this does not happen. I tried other dupe modes, but this was the only one that resolved the issue without duplicating components. This is also not the only component that has this mode even without any init args or InheritComponent, like Waddling. I do not know what would be a best way to fix this otherwise.

Reply:
On discussion with antur this seems fine but I'm going to have to think about some changes in the future with component code to handle this kind of usecase.